src/frontend/bundle-entry.jsx

```diff
@@ -21,8 +21,14 @@ const env = {
 
 const store = createStore(window.REDUX_STATE)
 const theme = window.THEME
+const adminLocale = window.REDUX_STATE.locale
 const currentLocale = JSON.parse(window.localStorage.getItem('locale'))
-const locale = currentLocale || window.REDUX_STATE.locale
+let locale = null
+if (currentLocale) {
+  locale = (adminLocale.language === currentLocale.language) ? adminLocale : currentLocale
+} else {
+  locale = adminLocale
+}
```

Comment:
I think this would be simpler but I suggest storing locales in i18n resources. Regarding `useSelector` comment, this variable is outside store provider so the store is here relevant to  `window.REDUX_STATE` or `store.getState()`.

```suggestion
let locale = adminLocale
if (currentLocale) {
  locale = adminLocale.language === currentLocale.language ? adminLocale : currentLocale
}
```